code/modules/tgs/v3210/api.dm

```diff
@@ -54,6 +54,8 @@
 	return ""
 
 /datum/tgs_api/v3210/proc/file2list(filename)
+	if(IsAdminAdvancedProcCall())
+		CRASH("Attempted to read file via admin call")
 	return splittext(trim_left(trim_right(file2text(filename))), "\n")
```

Comment:
Why is this bad?

Reply:
Because anyone with debug perms (on servers still using tgs3) can read the contents of any file they'd like. API keys, boot.ini, passwd, etc. not good xd